API.md

```diff
@@ -0,0 +1,1246 @@
+So this is a TransientError which could be retried after forcing "reconciliation" operation (which should recreate missing resource)
+
+**Parameters**
+
+-   `message` **[string][58]** 
+-   `extra` **[Object][57]** 
+
+### TransientError
+
+**Extends Error**
+
+This is a transient error related to either connectivity issues or temporary 3rd party API unavailability.
+
+When using `superagentErrorPlugin` it's returned by some errors out-of-the-box.
+
+**Parameters**
+
+-   `message` **[string][58]** 
+-   `extra` **[Object][57]** 
+
+## Helpers
+
+This is a set of additional helper functions being exposed at `req.hull.helpers`. They allow to perform common operation in the context of current request. They are similar o `req.hull.client.utils`, but operate at higher level, ensure good practises and should be used in the first place before falling back to raw utils.
+
+### handleExtract
+
+Helper function to handle JSON extract sent to batch endpoint
+
+**Parameters**
+
+-   `ctx` **[Object][57]** Hull request context
+-   `options` **[Object][57]** 
+    -   `options.body` **[Object][57]** request body object (req.body)
+    -   `options.batchSize` **[Object][57]** size of the chunk we want to pass to handler
+    -   `options.handler` **[Function][61]** callback returning a Promise (will be called with array of elements)
+    -   `options.onResponse` **[Function][61]** callback called on successful inital response
+    -   `options.onError` **[Function][61]** callback called during error
+
+Returns **[Promise][62]** 
+
+### requestExtract
+
+This is a method to request an extract of user base to be sent back to the Connector to a selected `path` which should be handled by `notifHandler`.
+
+**Parameters**
+
+-   `ctx` **[Object][57]** Hull request context
+-   `options` **[Object][57]**  (optional, default `{}`)
+    -   `options.segment` **[Object][57]**  (optional, default `null`)
+    -   `options.format` **[Object][57]**  (optional, default `json`)
+    -   `options.path` **[Object][57]**  (optional, default `batch`)
+    -   `options.fields` **[Object][57]**  (optional, default `[]`)
+    -   `options.additionalQuery` **[Object][57]**  (optional, default `{}`)
+
+**Examples**
+
+```javascript
+req.hull.helpers.requestExtract({ segment = null, path, fields = [], additionalQuery = {} });
+```
+
+Returns **[Promise][62]** 
+
+### updateSettings
+
+Allows to update selected settings of the ship `private_settings` object. This is a wrapper over `hullClient.utils.settings.update()` call. On top of that it makes sure that the current context ship object is updated, and the ship cache is refreshed.
+It will emit `ship:update` notify event.
+
+**Parameters**
+
+-   `ctx` **[Object][57]** The Context Object
+-   `newSettings` **[Object][57]** settings to update
+
+**Examples**
+
+```javascript
+req.hull.helpers.updateSettings({ newSettings });
+```
+
+Returns **[Promise][62]** 
+
+## Context
```

Comment:
missing introductory paragraph with a code snippet of all the possible items? like on https://github.com/hull/hull-node/blob/develop/API.md#hullconnector